constellations/orion/src/tests_integration/family_integration_test.go

```diff
@@ -0,0 +1,178 @@
+package integration_tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/stretchr/testify/assert"
+)
+
+type FamilySearchBody struct {
+	PrimaryEmail string
+}
+
+//create 1 family then get by id
+func Test_SearchFamilyById(t *testing.T) {
+	family1 := createFamily(1)
+	body1 := createJsonBody(&family1)
+	recorder1 := sendHttpRequest(t, http.MethodPost, "/api/families/create", body1)
+	assert.EqualValues(t, http.StatusOK, recorder1.Code)
+
+	recorder2 := sendHttpRequest(t, http.MethodGet, "/api/families/family/1", nil)
+	assert.EqualValues(t, http.StatusOK, recorder2.Code)
+
+	var family domains.Family
+	if err := json.Unmarshal(recorder2.Body.Bytes(), &family); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assertFamily(t, 1, family)
+
+	resetTable(t, domains.TABLE_FAMILIES)
+}
+
+// Test: Create 3 Families and search by pagination
+func Test_SearchFamilyByPrimaryEmail(t *testing.T) {
+	family1 := createFamily(1)
+	body1 := createJsonBody(&family1)
+	recorder1 := sendHttpRequest(t, http.MethodPost, "/api/families/create", body1)
+	assert.EqualValues(t, http.StatusOK, recorder1.Code)
+
+	// Create search body for HTTP request
+	familySearchBody := FamilySearchBody{
+		"john_smith@example.com",
+	}
+	marshal, err := json.Marshal(&familySearchBody)
+	if err != nil {
+		panic(err)
+	}
+	body := bytes.NewBuffer(marshal)
```

Comment:
Normally, I'm okay with creating a `domains` object and Marshalling it. But since this object is so small, maybe it's easier to do something like this:

```
body := strings.NewReader(`{
		"primaryEmail": "john@example.com",
	}`)
sendHttpRequest(......, body)
```